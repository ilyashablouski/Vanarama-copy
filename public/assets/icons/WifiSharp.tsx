import React, { FC, memo } from 'react';

const Svg: FC = memo(() => (
  <svg width="1em" height="1em" viewBox="0 0 512 512">
    <path
      d="M332.69,320a115,115,0,0,0-152.8,0"
      style={{
        fill: 'none',
        stroke: 'var(--class-color)',
        strokeLinecap: 'square',
        strokeLinejoin: 'round',
        strokeWidth: 42,
      }}
    />
    <path
      d="M393.74,259a201.26,201.26,0,0,0-274.92,0"
      style={{
        fill: 'none',
        stroke: 'var(--class-color)',
        strokeLinecap: 'square',
        strokeLinejoin: 'round',
        strokeWidth: 42,
      }}
    />
    <path
      d="M448,191.52a288,288,0,0,0-383.44,0"
      style={{
        fill: 'none',
        stroke: 'var(--class-color)',
        strokeLinecap: 'square',
        strokeLinejoin: 'round',
        strokeWidth: 42,
      }}
    />
    <path d="M300.67,384,256,433l-44.34-49a56.73,56.73,0,0,1,88.92,0Z" />
  </svg>
));

export default Svg;
