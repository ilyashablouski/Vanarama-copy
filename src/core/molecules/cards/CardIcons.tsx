import cx from 'classnames';
import React from 'react';
import Icon from '../../atoms/icon';
import Text from '../../atoms/text';

export type TIcon = {
  icon: React.ReactNode;
  label: string;
  index: string;
};

interface ICardIconsProps {
  className?: string;
  icons: TIcon[];
}

const CardIcons: React.FC<ICardIconsProps> = ({ className, icons }) => (
  <div className={cx('features', className)}>
    {/* At most only ever show 4 */}
    {icons.slice(0, 4).map(item => (
      <div key={`${item.label}_${item.index}`}>
        <Icon icon={item.icon} />
        <Text size="xsmall" color="dark">
          {item.label}
        </Text>
      </div>
    ))}
  </div>
);

export default CardIcons;
