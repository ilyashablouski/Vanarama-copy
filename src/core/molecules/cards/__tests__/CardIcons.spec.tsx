import React from 'react';
import { render } from '@testing-library/react';
import CardIcons, { TIcon } from '../CardIcons';
import BluetoothSharp from '../../../assets/icons/BluetoothSharp';
import CompassSharp from '../../../assets/icons/CompassSharp';
import SnowSharp from '../../../assets/icons/SnowSharp';
import WifiSharp from '../../../assets/icons/WifiSharp';

describe('<CardIcons />', () => {
  it('should render correctly', () => {
    // ARRANGE
    const icons: TIcon[] = [
      { icon: <SnowSharp />, label: 'Aircon', index: '1', name: 'Aircon' },
      {
        icon: <BluetoothSharp />,
        label: 'Bluetooth',
        index: '2',
        name: 'Bluetooth',
      },
      {
        icon: <CompassSharp />,
        label: 'Navigation',
        index: '3',
        name: 'Navigation',
      },
      { icon: <WifiSharp />, label: 'Sensors', index: '4', name: 'Sensors' },
    ];

    // ACT
    const wrapper = render(
      <CardIcons className="some-custom-class" icons={icons} />,
    );

    // ASSERT
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render with N/A when name undefined', () => {
    // ARRANGE
    const icons: TIcon[] = [
      { icon: <SnowSharp />, label: 'Aircon', index: '1', name: 'Aircon' },
      {
        icon: <BluetoothSharp />,
        label: 'Bluetooth',
        index: '2',
        name: '',
      },
      {
        icon: <CompassSharp />,
        label: 'Navigation',
        index: '3',
        name: '',
      },
      { icon: <WifiSharp />, label: 'Sensors', index: '4', name: '' },
    ];

    // ACT
    const wrapper = render(
      <CardIcons className="some-custom-class" icons={icons} />,
    );

    // ASSERT
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render no more than 4 icons', () => {
    // ARRANGE
    const icons: TIcon[] = [
      { icon: <SnowSharp />, label: 'Aircon', index: '1', name: 'Aircon' },
      {
        icon: <BluetoothSharp />,
        label: 'Bluetooth',
        index: '2',
        name: 'Bluetooth',
      },
      {
        icon: <CompassSharp />,
        label: 'Navigation',
        index: '3',
        name: 'Navigation',
      },
      { icon: <WifiSharp />, label: 'Sensors', index: '4', name: 'Sensors' },
      {
        icon: <SnowSharp />,
        label: 'I should not show!',
        index: '5',
        name: '',
      },
    ];

    // ACT
    const wrapper = render(
      <CardIcons className="some-custom-class" icons={icons} />,
    );

    // ASSERT
    expect(wrapper.container).toMatchSnapshot();
  });
});
