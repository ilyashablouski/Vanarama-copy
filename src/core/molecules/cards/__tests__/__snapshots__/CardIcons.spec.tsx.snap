// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardIcons /> should render correctly 1`] = `
<div
  className="features some-custom-class"
>
  <div
    key="Aircon_1"
  >
    <Icon
      icon={<Memo />}
    />
    <Memo(Text)
      color="dark"
      size="xsmall"
    >
      Aircon
    </Memo(Text)>
  </div>
  <div
    key="Bluetooth_2"
  >
    <Icon
      icon={<Memo />}
    />
    <Memo(Text)
      color="dark"
      size="xsmall"
    >
      Bluetooth
    </Memo(Text)>
  </div>
  <div
    key="Navigation_3"
  >
    <Icon
      icon={<Memo />}
    />
    <Memo(Text)
      color="dark"
      size="xsmall"
    >
      Navigation
    </Memo(Text)>
  </div>
  <div
    key="Sensors_4"
  >
    <Icon
      icon={<Memo />}
    />
    <Memo(Text)
      color="dark"
      size="xsmall"
    >
      Sensors
    </Memo(Text)>
  </div>
</div>
`;

exports[`<CardIcons /> should render no more than 4 icons 1`] = `
<div
  className="features some-custom-class"
>
  <div
    key="Aircon_1"
  >
    <Icon
      icon={<Memo />}
    />
    <Memo(Text)
      color="dark"
      size="xsmall"
    >
      Aircon
    </Memo(Text)>
  </div>
  <div
    key="Bluetooth_2"
  >
    <Icon
      icon={<Memo />}
    />
    <Memo(Text)
      color="dark"
      size="xsmall"
    >
      Bluetooth
    </Memo(Text)>
  </div>
  <div
    key="Navigation_3"
  >
    <Icon
      icon={<Memo />}
    />
    <Memo(Text)
      color="dark"
      size="xsmall"
    >
      Navigation
    </Memo(Text)>
  </div>
  <div
    key="Sensors_4"
  >
    <Icon
      icon={<Memo />}
    />
    <Memo(Text)
      color="dark"
      size="xsmall"
    >
      Sensors
    </Memo(Text)>
  </div>
</div>
`;
